### PR TITLE
add extrafi for optimism

### PR DESCRIPTION
**Pull request by yabirgb**
Related to #8013

- [x] Support Extra finance lending pools entering and exiting history event decoding
- [x] Support Extra finance lending pool rewards claiming in history event decoding
- [x] Support Extra finance farms in decoding events
- [x] Support Extra finance staking in decoding events
- [ ] Support balance detection in the lending pools and farms and staking.
- [ ] Support it in all evm chains they are deployed. Base is left

### CODE REVIEW

File: rotkehlchen/chain/optimism/modules/extrafi/decoder.py [hunk 1]
```diff
@@ -0,0 +1,371 @@
...
+                    'vault_id': hex_or_bytes_to_int(context.tx_log.topics[1]),
+                    'vault_position': hex_or_bytes_to_int(context.tx_log.topics[2]),
+                }
+
+        return DEFAULT_DECODING_OUTPUT
+
+    def _handle_farm_repayment(self, context: DecoderContext) -> DecodingOutput:
+        amount_0_repaid = hex_or_bytes_to_int(context.tx_log.data[32:64])
+        amount_1_repaid = hex_or_bytes_to_int(context.tx_log.data[64:96])
+        refund_event: EvmEvent | None = None  # refund event comes always first
```
[the remark this answers is not in the document]
Comment (yabirgb) [reply rotkehlchen/chain/optimism/modules/extrafi/decoder.py:299]: It was a mistake in my assumption. The refund event appears decoded before the spend event in the case of ethereum but in the case of a token repayment the spend happens before the receive. I've added a test case for this and adjusted the logic

File: rotkehlchen/chain/optimism/modules/extrafi/decoder.py [hunk 2]
```diff
@@ -0,0 +1,371 @@
...
+                event.counterparty = CPT_EXTRAFI
+                event.notes = f'Claim {amount} EXTRA from Extrafi'
+                break
+        else:
+            log.error(f'Could not match claim event of EXTRA in {context.transaction}')
+
+        return DEFAULT_DECODING_OUTPUT
+
+    def _handle_pool_rewards(self, context: DecoderContext) -> DecodingOutput:
+        """Handle rewards claimed for staking in pools"""
```
[the remark this answers is not in the document]
Comment (yabirgb) [reply rotkehlchen/chain/optimism/modules/extrafi/decoder.py:175]: The contracts create the position and stake it for what I saw. From the user side you don't see it but you are interacting with a staked position in velo. Will adjust

File: rotkehlchen/chain/optimism/modules/extrafi/decoder.py [hunk 3]
```diff
@@ -0,0 +1,371 @@
...
+                    asset_normalized_value(amount_0_received, event.asset) == event.balance.amount or  # type: ignore  # noqa: E501
+                    asset_normalized_value(amount_1_received, event.asset) == event.balance.amount  # type: ignore
+                )
+            ):
+                event.counterparty = CPT_EXTRAFI
+                event.event_type = HistoryEventType.WITHDRAWAL
+                event.event_subtype = HistoryEventSubType.REMOVE_ASSET
+                event.notes = f"Withdraw {event.balance.amount} {event.asset.symbol_or_name()} from Extrafi's farm"  # noqa: E501
+            elif (
+                event_direction == EventDirection.IN and
```
[the remark this answers is not in the document]
Comment (yabirgb) [reply rotkehlchen/chain/optimism/modules/extrafi/decoder.py:258]: it was a copy-paste mistake. Removed it

File: rotkehlchen/chain/optimism/modules/extrafi/decoder.py [hunk 4]
```diff
@@ -0,0 +1,350 @@
...
+                event.maybe_get_direction() == EventDirection.IN and
+                (
+                    asset_normalized_value(amount_0_received, event.asset) == event.balance.amount or  # type: ignore  # noqa: E501
+                    asset_normalized_value(amount_1_received, event.asset) == event.balance.amount  # type: ignore
+                )
+            ):
+                event.counterparty = CPT_EXTRAFI
+                event.event_type = HistoryEventType.WITHDRAWAL
+                event.event_subtype = HistoryEventSubType.REMOVE_ASSET
+                event.notes = f"Withdraw {event.balance.amount} {event.asset.symbol_or_name()} from Extrafi's farm"  # noqa: E501
```
Comment (LefterisJP) [inline rotkehlchen/chain/optimism/modules/extrafi/decoder.py:237]: shouldn't you break here? And also add an else: error log if this is not found?

File: rotkehlchen/chain/optimism/modules/extrafi/decoder.py [hunk 5]
```diff
@@ -0,0 +1,350 @@
...
+    from rotkehlchen.chain.evm.decoding.base import BaseDecoderTools
+    from rotkehlchen.chain.optimism.node_inquirer import OptimismInquirer
+    from rotkehlchen.types import ChecksumEvmAddress
+    from rotkehlchen.user_messages import MessagesAggregator
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+CPT_EXTRAFI: Final = 'extrafi'
```
Comment (LefterisJP) [inline rotkehlchen/chain/optimism/modules/extrafi/decoder.py:42]: All these constants should go in a constants file. First a generic evm one and then per chain one. you can do it the PR where you add it per-chain

**Review by LefterisJP — COMMENTED** [review]
Some more comments

**Review by LefterisJP — APPROVED** [review]
LGTM

**Pull request merged**